lib/semian/trilogy_adapter.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require "semian/adapter"
+require "activerecord-trilogy-adapter"
+require "active_record/connection_adapters/trilogy_adapter"
+
+module ActiveRecord
+  module ConnectionAdapters
+    class TrilogyAdapter
+      ActiveRecord::ActiveRecordError.include(::Semian::AdapterError)
+
+      class SemianError < StatementInvalid
+        def initialize(semian_identifier, *args)
+          super(*args)
+          @semian_identifier = semian_identifier
+        end
+      end
+
+      ResourceBusyError = Class.new(SemianError)
+      CircuitOpenError = Class.new(SemianError)
+    end
+  end
+end
+
+module Semian
+  module TrilogyAdapter
```

Comment:
I think we should call this `Semian::ActiveRecordTrilogyAdapter`.

Reply:
That a good point.

https://github.com/Shopify/semian/pull/468/files#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55fR22

I wonder what will be in future name of adapter when we implement semian for pure trilogy.